lib/slack/rtm.ex

```diff
@@ -10,15 +10,25 @@ defmodule Slack.Rtm do
   @moduledoc false
 
   def start(token) do
-    url = Application.get_env(:slack, :url, "https://slack.com") <> "/api/rtm.start?token="
+    slack_url(token)
+    |> HTTPoison.get()
+    |> handle_response()
+  end
 
-    case HTTPoison.get(url <> token) do
-      {:ok, %HTTPoison.Response{body: body}} ->
-        case JSX.decode(body, [{:labels, :atom}]) do
-          {:ok, json}       -> {:ok, json}
-          {:error, reason}  -> {:error, %JSX.DecodeError{reason: reason, string: body}}
-        end
-      {:error, reason} -> {:error, reason}
+  defp handle_response({:ok, %HTTPoison.Response{body: body}}) do
+    with {:ok, json} <- JSX.decode(body, [{:labels, :atom}]),
+      %{ok: true} = json do
+        {:ok, json}
+    else
+      {:error, reason} -> {:error, %JSX.DecodeError{reason: reason, string: body}}
```

Comment:
hmm, i actually need to match on `%{error: reason}` as well for the case of `%{ok: false}` on line 20